crates/agent/src/profile_selector.rs

```diff
@@ -123,7 +126,15 @@ impl ProfileSelector {
             let fs = self.fs.clone();
             let thread_store = self.thread_store.clone();
             let profile_id = profile_id.clone();
+            let profile = profile.clone();
+
+            let thread = self.thread.clone();
+
             move |_window, cx| {
+                thread.update(cx, |thread, cx| {
+                    thread.set_configured_profile(Some(profile.clone()), cx);
+                });
+
                 update_settings_file::<AssistantSettings>(fs.clone(), cx, {
```

Comment:
I think if this is going to be per-thread state going forward we shouldn't update the settings file here